src/lib/mina_metrics/prometheus_metrics/mina_metrics.ml

```diff
@@ -459,49 +459,68 @@ module Network = struct
       let help = "# of blocks received" in
       Counter.v "received" ~help ~namespace ~subsystem
 
-    module Validation_time =
-      Moving_time_average
-        (struct
-          include Delay_time_spec
+    let validation_time_is_initialized = ref false
 
-          let subsystem = subsystem
+    let processing_time_is_initialized = ref false
 
-          let name = "validation_time"
+    let rejection_time_is_initialized = ref false
 
-          let help =
-            "average time, in ms, for blocks to be validated and rebroadcasted"
-        end)
-        ()
+    module Gauge_map = Metric_map (struct
+      type t = Gauge.t
 
-    module Processing_time =
-      Moving_time_average
-        (struct
-          include Delay_time_spec
+      let subsystem = subsystem
 
-          let subsystem = subsystem
+      let v = Gauge.v
+    end)
 
-          let name = "processing_time"
+    let block_table = Gauge_map.of_alist_exn []
 
-          let help =
-            "average time, in ms, for blocks to be accepted after the OCaml \
-             process receives it"
-        end)
-        ()
+    module Validation_time (Context : CONTEXT) = Moving_time_average (struct
+      include Delay_time_spec (Context)
 
-    module Rejection_time =
-      Moving_time_average
-        (struct
-          include Delay_time_spec
+      let subsystem = subsystem
 
-          let subsystem = subsystem
+      let name = "validation_time"
 
-          let name = "rejection_time"
+      let help =
+        "average time, in ms, for blocks to be validated and rebroadcasted"
 
-          let help =
-            "average time, in ms, for blocks to be rejected after the OCaml \
-             process receives it"
-        end)
-        ()
+      let v = Gauge_map.add block_table ~name ~help
+
+      let is_initialized = validation_time_is_initialized
+    end)
+
+    module Processing_time (Context : CONTEXT) = Moving_time_average (struct
```

Comment:
I'm not sure that this refactoring of metrics code is the best approach.

AFAIU, block window is only needed to compute intervals for loops that update moving average metrics.

What is we could leave the metrics alone (same code as it was, no `CONTEXT`) and just have each metric to expose an `initialize : block_window_ms:Time.Span.t -> unit` function which all are called from somewhere on the top.

Before this function is called, metrics wouldn't be maintained, but this is a neglible effect.